### PR TITLE
VLCServerListViewController: move connect cell into sectionHeader

### PR DESCRIPTION
**Pull request by carolanitz**
### Checklist
- [x] I've run `bundle exec fastlane test` from the root directory to see all new and existing tests pass
- [x] I've followed the [vlc-ios code style](Docs/CodingStyle.md) and run `bundle exec fastlane lint` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/videolan/vlc-ios#contribute)
- [x] I've updated the documentation if necessary.

### Description

still not perfect and I might have to do some cleanup but feedback welcome

screen shot 2018-07-04 at 4 04 11 pm

### CODE REVIEW

**Comment by carolanitz** [comment]
The baseline of both labels should be aligned and they shouldn't be both centered on the header. The background is also still the wrong color and in figma the separator is a bit indented. It might also make sense to cache the VLCNetworkLoginViewController to not create it every time

File: Resources/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -124,6 +124,8 @@
 
 "LOCAL_NETWORK" = "Local Network";
 "CONNECT_TO_SERVER" = "Connect to Server";
+"CONNECT" = "Connect";
+"FILE_SERVER" = "File Servers";
```
Comment (Mikanbu) [inline Resources/en.lproj/Localizable.strings]: Do we need to add new strings? For `Connect` could we use `"BUTTON_CONNECT" = "Connect";` ?

File: Sources/VLCFileServerSectionTableHeaderView.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*****************************************************************************
+ * VLCFileServerSectionTableHeaderView.m
+ * VLC for iOS
+ *****************************************************************************
+ * Copyright (c) 2018 VideoLAN. All rights reserved.
+ * $Id$
+ *
+ * Author: Carola Nitz <caro # videolan.org>
+
```
Comment (Mikanbu) [inline Sources/VLCFileServerSectionTableHeaderView.swift]: Extra line

File: Sources/VLCFileServerSectionTableHeaderView.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        super.setupUI()
+        self.textLabel?.text = NSLocalizedString("FILE_SERVER", comment: "")
+    }
+
+    //Before layoutSubviews textlabel doesn't have a superview
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        if layoutConstraints == nil {
+            layoutConstraints = [
+                connectButton.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: -20),
```
Comment (Mikanbu) [inline Sources/VLCFileServerSectionTableHeaderView.swift]: I think we can remove the `self.`

File: Sources/VLCFileServerSectionTableHeaderView.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                connectButton.firstBaselineAnchor.constraint(equalTo: self.textLabel!.firstBaselineAnchor)
```
Comment (Mikanbu) [inline Sources/VLCFileServerSectionTableHeaderView.swift]: ^ same

File: Sources/VLCFileServerSectionTableHeaderView.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    }
+    @objc func connectButtonDidPress() {
+        delegate?.connectToServer()
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        //Text gets set to nil in prepareForReuse so we set it again
+        self.textLabel?.text = NSLocalizedString("FILE_SERVER", comment: "")
```
Comment (Mikanbu) [inline Sources/VLCFileServerSectionTableHeaderView.swift]: ^ same

File: Sources/VLCSectionTableHeaderView.swift [hunk 6]
```diff
@@ -0,0 +1,54 @@
+/*****************************************************************************
+ * VLCSectionTableHeaderView.m
+ * VLC for iOS
+ *****************************************************************************
+ * Copyright (c) 2018 VideoLAN. All rights reserved.
+ * $Id$
+ *
+ * Author: Carola Nitz <caro # videolan.org>
+
```
Comment (Mikanbu) [inline Sources/VLCSectionTableHeaderView.swift]: Extra line

File: Sources/VLCFileServerSectionTableHeaderView.swift [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        connectButton.titleLabel?.textColor = PresentationTheme.current.colors.orangeUI
+        connectButton.translatesAutoresizingMaskIntoConstraints = false
+        connectButton.addTarget(self, action: #selector(connectButtonDidPress), for: .touchUpInside)
+        contentView.addSubview(connectButton)
+        return connectButton
+    }()
+
+    override func setupUI() {
+        super.setupUI()
+        self.textLabel?.text = NSLocalizedString("FILE_SERVER", comment: "")
```
Comment (Mikanbu) [inline Sources/VLCFileServerSectionTableHeaderView.swift]: I think we can remove the `self.`

File: Sources/VLCSectionTableHeaderView.swift [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    @objc func updateTheme() {
+        contentView.backgroundColor = PresentationTheme.current.colors.background
+        separator.backgroundColor = PresentationTheme.current.colors.separatorColor
+        textLabel?.textColor = PresentationTheme.current.colors.cellTextColor
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        textLabel?.font = UIFont.systemFont(ofSize: 24, weight: .semibold)
```
Comment (Mikanbu) [inline Sources/VLCSectionTableHeaderView.swift]: Should we consider declaring this constant at the top of the file?

File: Resources/en.lproj/Localizable.strings (hunk 1 above)
Comment (carolanitz) [reply Resources/en.lproj/Localizable.strings]: yes we need to

File: SharedSources/PresentationTheme.swift [hunk 9]
```diff
@@ -33,7 +33,7 @@ extension Notification.Name {
     let sectionHeaderTintColor: UIColor
     let settingsBackground: UIColor
     let settingsCellBackground: UIColor
-    let settingsSeparatorColor: UIColor
+    let separatorColor: UIColor
     let mediaCategorySeparatorColor: UIColor
```
[the remark this answers is not in the document]
Comment (carolanitz) [reply SharedSources/PresentationTheme.swift]: no this is part of this cell

[the remark this answers is not in the document]
Comment (Mikanbu) [reply SharedSources/PresentationTheme.swift]: but doesn't it touch the SettingView as well?

File: Sources/VLCSectionTableHeaderView.swift (hunk 8 above)
Comment (carolanitz) [reply Sources/VLCSectionTableHeaderView.swift]: maybe we should have the fonts somewhere in general

**Comment by carolanitz** [comment]
merged with 1b2fe4bcffcb42111f9e83ff099f5178b88362c9

**Pull request closed without merging**